src/phoenix/server/bearer_auth.py

```diff
@@ -0,0 +1,68 @@
+from abc import ABC
+from typing import Any, Awaitable, Callable, Optional, Protocol, Tuple
+
+import grpc
+from grpc_interceptor import AsyncServerInterceptor
+from grpc_interceptor.exceptions import Unauthenticated
+from starlette.authentication import AuthCredentials, AuthenticationBackend, BaseUser
+from starlette.requests import HTTPConnection
+
+from phoenix.auth import PHOENIX_ACCESS_TOKEN_COOKIE_NAME, Claim, ClaimStatus, Token
+
+
+class CanReadToken(Protocol):
+    async def read(self, token: Token) -> Claim: ...
+
+
+class HasTokenStore(ABC):
+    def __init__(self, token_store: CanReadToken) -> None:
+        super().__init__()
+        self._token_store = token_store
+
+
+class BearerTokenAuthBackend(HasTokenStore, AuthenticationBackend):
```

Comment:
It's not super clear to me - we are using bearer for the access tokens correct? Doesn't this overlap with @axiomofjoy's work?

Reply:
Yes, this is the `Bearer` scheme, which i thought we had decided on during stand up. sorry if i misunderstood. 

On the other hand, the `Bearer` scheme seems be to the most appropriate based on literature.

Reply:
I guess I'm getting a bit confused on the terminology. So since the permissions of the keys are "dynamic" - we consider even our API keys "bearer tokens" - is that the rationale? 

> We can differentiate between an API key and a token with the following definitions:

API key – A value provided by code when calling an API to identify and authorize the caller. It is intended to be used programmatically and is often a long string of letters and numbers.
Token – A piece of data that represents a user session or specific privileges. Used by individual users for a limited period of time.

Reply:
yea, conceptually they're both just tokens if we separate `Authentication` from `Access control`, resulting in a "signed pointer".